\documentclass[oupdraft]{bio}
% \usepackage[colorlinks=true, urlcolor=citecolor, linkcolor=citecolor, citecolor=citecolor]{hyperref}
\usepackage{url}

% Add history information for the article if required
\history{Received XXX;
revised XXX;
accepted for publication XXX}


\newcommand{\todo}[1]{\textbf{[TODO: #1]}}
\newcommand{\note}[1]{\textbf{[NOTE: #1]}}
\newcommand{\fixme}[1]{\textbf{[FIXME: #1]}}

\begin{document}

% Title of paper
\title{My very cool title}

% List of authors, with corresponding author marked by asterisk
\author{Nelle Varoquaux$^\ast$,\\[4pt]
% Author addresses
\textit{Department of Statistics, University of California, Berkeley \\
Berkeley Data Science Fellow}
\\[2pt]
% E-mail address for correspondence
{eaheron@tcd.ie}}

% Running headers of paper:
\markboth%
% First field is the short list of authors
{Nelle Varoquaux}
% Second field is the short title of the paper
{XXX}

\maketitle

% Add a footnote for the corresponding author if one has been
% identified in the author list
\footnotetext{To whom correspondence should be addressed.}

\begin{abstract}
{My very cool abstract.}
{Hi-C; three-dimensional structure; and other cool stuff.
}
\end{abstract}


\section{Introduction}
\label{sec1}


\section{Methods}
\label{sec2}

Let us first introduce some notations. Hi-C data can be summarized as an $n
\times n$ symmetric contact count matrix, where each row and column corresponds to a
genomic window (or loci), and each entry $c_{ij}$ refers to the number of time
$i$ and $j$ have been seen interacting with one another. The matrix is hollow

Hi-C data suffers from technical and biological biases that need to be
corrected. \fixme{blha} 


\subsection{Analysing contact count data}

- Normalization
- Quality assessment
- Enrichment of counts
- Anything else?
    - Did we do a compartment analysis?
- Can we extend it?
  - Fit-HicÂ ?

\subsection{3D structure inference}

A crucial step of our analysis of the \textit{P. falciparum} genome relied on
visual introspection of a consensus 3D model of the genome. Being able to
quickly visualize how chromosome folded, where families of genes lied in the
3D space or whether regions where enriched helped us integrate with other
sources of data.

Briefly, we cast the problem of finding a consensus 3D structure $X \in R^{n
\times n}$ as an optimization problem, either as minimizing a cost function,
(often measuring the error between euclidean distance of $\mathbf{X}$ and a
series of ``wish distances'') or maximizing a likelihood. The optimization
problem can either be unconstrained or constrained to reflect some prior
knowledge on the structure (size and shape of the nucleas, distances between
loci, etc).

We added constraints to our model based on prior knowledge, such as the size
of the nucleus, or how far apart two adjacent beads should lie, though we
showed these constraints where well satisfied even when performing an
unconstrained optimized. In this section, I will review the inference method
we used in \citet{ay:three-dimensional} as well as two newer methods, based on
well-grounded statistical method.

\subsubsection{Multidimensional scaling}

\begin{equation}\label{eq:mds1} 
\renewcommand{\arraystretch}{2} 
\begin{array}{ccll} 
\underset{\mathbf{X}}{\text{minimize}} & & 
\underset{(i,j) \in \mathcal{D}}{\sum} w_{ij} \big(d_{ij}(\mathbf{X}) - \delta_{ij}\big)^2
\,,&\\ 
\end{array} 
\end{equation}

where $w_{ij}$ corresponds to weight.


\subsubsection{Modeling contact counts as a Poisson distribution}

\subsubsection{Modeling contact counts as a Negative Binomial distribution}


\subsubsection{Structure stability}
- Structure stability
- at least 100 iterations
- maybe compare results with chromSDE, PCA \& ShRec3D 
- 
\subsubsection{Volume exclusion models}

\subsection{Integrative analysis of gene expression and 3D structure using KernelCCA}

- Can we extend this to other data types? Chip-seq data available from our
review? Can the results be interpretable?

\subsection{Differential analysis}
\note{XXX only possible if Bunnik 2017 paper published.}

\section{Results}
\label{sec3}

\subsection{Simulation study application}

\section{Discussion}
\label{sec4}



\section{Software}
\label{sec5}

Software in the form of R code, together with a sample
input data set and complete documentation is available on
request from the corresponding author (eaheron@tcd.ie).


\section{Supplementary Material}
\label{sec6}

Supplementary material is available online at
% \href{http://biostatistics.oxfordjournals.org}%
% {http://biostatistics.oxfordjournals.org}.
\url{http://biostatistics.oxfordjournals.org}.


\section*{Acknowledgments}

The authors thank XXX

Funding for the project was provided by BIDS.
{\it Conflict of Interest}: None declared.


\bibliographystyle{biorefs}
\bibliography{refs}


\begin{figure}[!p]
\centering
%\includegraphics{fig1}
\caption{}
\label{Fig1}
\end{figure}

\end{document}
