\documentclass[oupdraft]{bio}
% \usepackage[colorlinks=true, urlcolor=citecolor, linkcolor=citecolor, citecolor=citecolor]{hyperref}
\usepackage{url}

% Add history information for the article if required
\history{Received XXX;
revised XXX;
accepted for publication XXX}


\newcommand{\todo}[1]{\textbf{[TODO: #1]}}
\newcommand{\note}[1]{\textbf{[NOTE: #1]}}
\newcommand{\fixme}[1]{\textbf{[FIXME: #1]}}

\begin{document}

% Title of paper
\title{Analysing the three-dimensional structure of \em{P. falciparum}}

% List of authors, with corresponding author marked by asterisk
\author{Nelle Varoquaux$^\ast$,\\[4pt]
% Author addresses
\textit{Department of Statistics, University of California, Berkeley \\
Berkeley Data Science Fellow}
\\[2pt]
% E-mail address for correspondence
{nelle@berkeley.edu}}

% Running headers of paper:
\markboth%
% First field is the short list of authors
{Nelle Varoquaux}
% Second field is the short title of the paper
{XXX}

\maketitle

% Add a footnote for the corresponding author if one has been
% identified in the author list
\footnotetext{To whom correspondence should be addressed.}

\begin{abstract}
{My very cool abstract.}
{Hi-C; three-dimensional structure; and other cool stuff.
}
\end{abstract}


\section{Introduction}
\label{sec1}


\section{Methods}
\label{sec2}


\subsection{Analysing contact count data}

- Normalization
- Quality assessment
- Enrichment of counts
- Anything else?
    - Did we do a compartment analysis?
- Can we extend it?
  - Fit-HicÂ ?

\subsection{Inferring the 3D structures}

An important part of analysing the 3D structure of the genome of \textit{P.
falciparum} relied on having consensus models for each stage of the life
cycle. The method presented in \citet{ay:three-dimensional} relied on an
approach called ``multidimensional scaling'' (MDS), widely used in the
literature for the purpose of 3D structure inference
\citep{duan:three-dimensional, varoquaux:statistical, bau:, tanizawa:}. Each chromosomes is
modeled a series of beads, spaced ~10~kb apart. MDS-based method consists of
two steps: first convert the
contact counts $c_{ij}$ into ``wish-distances'' $\delta_{ij}$; then attempt
to place each bead in 3D such that the euclidean distances $d_{ij}$ separating
beads $i$ and $j$ matches as closely as possible the wish distance
$\delta_{ij}$.

While these approaches are widely used, since better motivated and more
flexible models to infer 3D consensus structures from contact counts have been
proposed. 

\subsubsection{Constrained multi-dimensional scaling}

The method presented in \citet{ay:three-dimensional} relied on an
approach called ``multidimensional scaling'' (MDS), widely used in the
literature for the purpose of 3D structure inference
\citep{duan:three-dimensional, varoquaux:statistical, bau:, tanizawa:}. Each chromosomes is
modeled a series of beads, spaced ~10~kb apart. MDS-based method consists of
two steps: first convert the
contact counts $c_{ij}$ into ``wish-distances'' $\delta_{ij}$; then attempt
to place each bead in 3D such that the euclidean distances $d_{ij}$ separating
beads $i$ and $j$ matches as closely as possible the wish distances
$\delta_{ij}$.

\paragraph{Wish distances}

A crucial step of MDS-based method rely on finding a good set of
``wish-distances'' from Hi-C contact count data.

\paragraph{Optimization}

Given the resulting wish-distances, we cast the inference of the consensus
structure $\mathbf{X}^{3 \times n}$ as the following constrained weighted MDS:


\begin{equation}\label{eq:mds1} 
\renewcommand{\arraystretch}{2} 
\begin{array}{ccll} 
\underset{\mathbf{X}}{\text{minimize}} & & 
\underset{(i,j) \in \mathcal{D}}{\sum} \frac{1}{\delta^2_{ij}}
\big(d_{ij}(\mathbf{X}) - \delta_{ij}\big)^2 & \\
\text{subject to} & & d_{ij} \leq r^2_\text{max} \quad \forall i, j &\,,\\
& & d_{i, i+1} \leq b_\text{max} \quad i = \{i:n|\text{chr}_i = \text{chr}_{i+1}\,,&\\ 
\end{array} 
\end{equation}

The various constraints corresponds to {\em prior} knowledge on the 3D
structure, and vary across organisms, time-points of the life cycle and
cell-type. For {\em P. falciparum}, the constraints are as follow.

\begin{itemize}
\item All beads lie in a spherical nucleus centered on the origin and of a
certain size.
\item Two adjacent beads must not be too far apart. Leveraging the work of
\citet{berger}, we assume that $b^\text{max} = 91~nm$
\end{itemize}

Note that in practice, the constraints are mostly intrinsic to the data and
thus can be dropped.

\subsubsection{Modeling contact counts as a Poisson distribution}

\subsubsection{Modeling contact counts as a Negative Binomial distribution}


\subsubsection{Structure stability}
- Structure stability
- at least 100 iterations
- maybe compare results with chromSDE, PCA \& ShRec3D 
- 

\subsubsection{Volume exclusion models}


\subsection{Integrative analysis of gene expression and 3D structure using KernelCCA}

- Can we extend this to other data types? Chip-seq data available from our
review? Can the results be interpretable?

\subsection{Differential analysis}
\note{XXX only possible if Bunnik 2017 paper published.}

\section{Results}
\label{sec3}

\section{Discussion}
\label{sec4}



\section{Software}
\label{sec5}

Software in the form of R code, together with a sample
input data set and complete documentation is available on
request from the corresponding author (eaheron@tcd.ie).


\section{Supplementary Material}
\label{sec6}

Supplementary material is available online at
% \href{http://biostatistics.oxfordjournals.org}%
% {http://biostatistics.oxfordjournals.org}.
\url{http://biostatistics.oxfordjournals.org}.


\section*{Acknowledgments}

The authors thank XXX

Funding for the project was provided by BIDS.
{\it Conflict of Interest}: None declared.


\bibliographystyle{biorefs}
\bibliography{refs}


\begin{figure}[!p]
\centering
%\includegraphics{fig1}
\caption{}
\label{Fig1}
\end{figure}

\end{document}
