%\documentclass[2columns]{article}

\documentclass[letterpaper,12pt]{article}
\usepackage{url}
\usepackage{graphicx}
\usepackage{url}
\usepackage{wrapfig}
\usepackage[normalem]{ulem}
\usepackage{float}
\usepackage{xr}

\usepackage{amsmath,amssymb,amsfonts,amscd,latexsym}
\usepackage{tabularx}
\usepackage{multirow}
\usepackage{multicol}
\usepackage{eurosym}
\usepackage{geometry}
\usepackage{fancyhdr}
\usepackage{enumitem}
\usepackage{xspace}

% Two column foonotes
\usepackage{dblfnote}


%% This is the recommended preamble for your document.

%% Load De Gruyter specific settings 
\usepackage{dgjournal}          

%% The mathptmx package is recommended for Times compatible math symbols.
%% Use mtpro2 or mathtime instead of mathptmx if you have the commercially
%% available MathTime fonts.
%% Other options are txfonts (free) or belleek (free) or TM-Math (commercial)
\usepackage{mathptmx}

%% Use the graphics package to include figures
\usepackage{graphics}

%% Use natbib with these recommended options
\usepackage[authoryear,comma,longnamesfirst,sectionbib]{natbib} 


%\externaldocument{supp}


\newcommand{\todo}[1]{\textbf{[TODO: #1]}}
\newcommand{\note}[1]{\textbf{[NOTE: #1]}}
\newcommand{\fixme}[1]{\textbf{[FIXME: #1]}}

\begin{document}

% Title of paper
\title{Analysing the three-dimensional structure of P. falciparum}
\author{Nelle Varoquaux}

%\maketitle


\begin{abstract}

\end{abstract}


\section{Introduction}
\label{sec1}


\section{Methods}
\label{sec2}


\subsection{Analysing contact count data}

- Normalization
- Quality assessment
- Enrichment of counts
- Anything else?
    - Did we do a compartment analysis?
- Can we extend it?
  - Fit-Hic ?

\subsection{Inferring the 3D structures}

An important part of analysing the 3D structure of the genome of \textit{P.
falciparum} relied on having consensus models for each stage of the life
cycle. The method presented in \citet{ay:three-dimensional} relied on an
approach called ``multidimensional scaling'' (MDS), widely used in the
literature for the purpose of 3D structure inference
\citep{duan:three-dimensional, varoquaux:statistical, bau:, tanizawa:}. Each chromosomes is
modeled a series of beads, spaced ~10~kb apart. MDS-based method consists of
two steps: first convert the
contact counts $c_{ij}$ into ``wish-distances'' $\delta_{ij}$; then attempt
to place each bead in 3D such that the euclidean distances $d_{ij}$ separating
beads $i$ and $j$ matches as closely as possible the wish distance
$\delta_{ij}$.

\subsubsection{Constrained multi-dimensional scaling}

The method presented in \citet{ay:three-dimensional} relied on an
approach called ``multidimensional scaling'' (MDS), widely used in the
literature for the purpose of 3D structure inference
\citep{duan:three-dimensional, varoquaux:statistical, bau:, tanizawa:}. Each chromosomes is
modeled a series of beads, spaced ~10~kb apart. MDS-based method consists of
two steps: first convert the
contact counts $c_{ij}$ into ``wish-distances'' $\delta_{ij}$; then attempt
to place each bead in 3D such that the euclidean distances $d_{ij}$ separating
beads $i$ and $j$ matches as closely as possible the wish distances
$\delta_{ij}$.

\paragraph{Optimization}

Given the resulting wish-distances, we cast the inference of the consensus
structure $\mathbf{X}^{3 \times n}$ as the following constrained weighted MDS:


\begin{equation}\label{eq:mds1} 
\renewcommand{\arraystretch}{2} 
\begin{array}{ccll} 
\underset{\mathbf{X}}{\text{minimize}} & & 
\underset{(i,j) \in \mathcal{D}}{\sum} \frac{1}{\delta^2_{ij}}
\big(d_{ij}(\mathbf{X}) - \delta_{ij}\big)^2 & \\
\text{subject to} & & d_{ij} \leq r^2_\text{max} \quad \forall i, j &\,,\\
& & d_{i, i+1} \leq b_\text{max} \quad i = \{i:n|\text{chr}_i = \text{chr}_{i+1}\,,&\\ 
\end{array} 
\end{equation}

The various constraints corresponds to {\em prior} knowledge on the 3D
structure, and vary across organisms, time-points of the life cycle and
cell-type. For {\em P. falciparum}, the constraints are as follow.

\begin{itemize}
\item All beads lie in a spherical nucleus centered on the origin and of a
certain size.
\item Two adjacent beads must not be too far apart. Leveraging the work of
\citet{berger:high TOVERIFY}, we assume that $b^\text{max} = 91~nm$
\end{itemize}

\paragraph{Wish distances}

A crucial step of MDS-based method rely on finding a good set of
``wish-distances'' from Hi-C contact count data. While several approach have
been used in the litterature, most leverage bio-physics relationships either
between the genomic distnaces $s$ and the expected count, or the relationship
between the euclidean distance $d$ and the expected
count. \citet{ay:three-dimensional} observe that {\em P. falciparum}'s DNA
behave like a fractal globule polymer. \citet{grosberg:fractal} show fractal
globul polymer exhibits a well-defined relationship between its genomic length
and physical distance $d$:

$$d\sim s^{1/3}$$

The relationship between genomic distance $s$ and $c$ can be derived from the
data, by fitting a linear model between the log expected count and the genomic
distances and extracting the slope coefficient $\alpha$. We thus obtain, up to
a scaling factor $\beta$ a function to map contact counts to wish distances:

\begin{equation}
\label{eq:count-to-distance}
\delta_{ij} = \beta c^{\alpha/3}
\end{equation}

The scaling factor depends both on the sequencing depth of the libraries, but
also on the nucleus size of the organims. \citet{ay:three-dimensional} infers
3D models for a range of parameters and keeps the models that are the most
plausible, under the assumption that beads should be as uniformely distributed
in the nucleus as possible. 0-contact counts can either be ignored, or mapped
to the largest distance possible. Note that for high coverage data set, the
amount of non-interacting beads is sufficiently small that including those or
not in the optimization will make very little difference.

\subsubsection{Reproducing the results using a simpler model}

Two simplifications can be brought to the method proposed by
\citet{ay:three-dimensional}. 

First, the ``count-to-distance'' relationship is based on two steps: (1) find
the relationship between genomic distances and counts; (2) from the
coefficient, deduce what type of polymer the DNA of this particular organism
is closest two (fractal globule or equilibrium); (3) use the relationship
between genomic distance $s$ and euclidean distance $d$ for that particular
type of polymer. Yet, it is likely that the probability of contact count
between two elements depends mostly on the average distance separating the two
beads. As noted in \citet{varoquaux:statistical}, while the relationship
between both genomic distance $s$ and expected contact count $c$ and the
relationship between genomic distance $s$ and euclidean distance $c$,
combining both yields the same count-to-distance function for both polymer
type:

\begin{equation}
\delta_{ij} = \beta c^{-1/3}
\end{equation}

While the difference may seem subtile, fitting the linear model to find
$\alpha$ is not as straightforward as it seems, as the range of genomic
distance used can have a large impact on the final coefficient.

The second simplification is to perform an unconstraint optimization. Indeed,
in practice, the constraints are fulfilled naturally by the data and thus can
be dropped from the optimization. In addition of drastically reducing the
computation time, this allows to use generic publicly available software such
as chromSDE (which implements a variant of the model defined here), or
Pastis-MDS (which implements exactly the unconstrained model defined here).

\subsubsection{Structure stability}

Both the constrained and unconstained problem stated above are not convex. The
reader may well ask how sensitive the result is to initialize. In short, it is
not. To study the stability of the results with respect to the initialization,
I perform for each stage presented here 1000 optimization, starting from
different initialization. 

The first question that arise is simply: are structures from the same time
points but from a different start ponit more alike than structures from
different time points? To answer this question, I compute for each structure
and each time point a feature matrix, composed of a subsampled pairwise
euclidean distances of each structure. The reason for subsampling is simply so
that the resulting feature matrix fits in memory on a laptop. I then compute a
stochastic PCA on this feature matrix and plot the projection in 2D space
using the 3 first components. The results clearly show that structures from
the same time point cluster strongly. Repeating this experiment on singly
chromosomes yields the same result.

The second question one can ask is how do the structures differ. Tackling this
question is very challenging, but can be reformulated in a much easier way.
Are the hallmarks of interest conserved across structures of a same stage?
\citet{ay:three-dimensional} and \citet{lemieux:genome-wide} both identified
{\em P. falciparum} folded in very specific ways, with VRSSM genes highly
interacting. \citet{ay:three-dimensional} also observed strong clustering of
the centromeres, and enrichment in interaction at the telomeres. These
observation can lead to a rigourous approach at identifying whether families
of loci clustered in the structures and even quantifying this in a rigourous
manner.

A number of methods have been proposed to accomplish such tasks, which I will
review in this section.

\subsubsection{Volume exclusion models}

\citet{tjong:physical} showed \textit{S. cerevisiae}'s structure can be mostly
explained by a large set of constrained yet random population of structures
following a simple volume exclusion model. We repeated the experiment on {\em
P. falciparum}, and showed that, despite recovering similar results on {\em S.
cerevisiae}, the same set of constraints is not sufficient to explain {\em P.
falciparum}'s Hi-C contact map. 

Briefly, let us recall the methodology presented by \citet{tjong:physical}. We
construct a population of three-dimensional structures by modeling chromosomes
as random configuration of beads-on-a-string, subjected to the following constraints.

\begin{itemize}
\item \textbf{Adjacent Beads Constraints} Two adjacent beads spaced 3.2kb apart are constrained to be 30nm apatr.
\item \textbf{Volume Exclusion Constraints} Each pairs of beads are spaced at minimum XX apart
\item \textbf{Nucleus Constraint} All beads lie in a spherical nucleus of specific
radius (XXX)
\item \textbf{Centromeres Constraints} Centromeres are localized in a small sphere of
50nm abutting the nuclear envelop.
\item \textbf{Telomere Constraints} Telomeres are located at most 50nm from the nuclear envelop.
\end{itemize}

\citet{tjong:physical} then formulates an optimization problem that includes
in addition to all the aforementioned constraints a penalty term on three
adjacent beads accounting for chromatin stiffness.

$$
\frac{1}{2} FIXME
$$

We use IMP to generate a population of structures.

\paragraph{Adding constraints on VRSM genes}

\subsection{Integrative analysis of gene expression and 3D structure using KernelCCA}

\citet{ay:three-dimensional}  exploited the 3D models to relate structure to
gene expression. The authors rely on a technic called ``kernelCCA'' with
attempts to find components on different view of the data in such a way that
these are coherent with one another.

- Can we extend this to other data types? Chip-seq data available from our
review? Can the results be interpretable?


\section{Results}
\label{sec3}

\section{Discussion}
\label{sec4}



\section{Software}
\label{sec5}

Software in the form of R code, together with a sample
input data set and complete documentation is available on
request from the corresponding author (eaheron@tcd.ie).


\section{Supplementary Material}
\label{sec6}

Supplementary material is available online at
% \href{http://biostatistics.oxfordjournals.org}%
% {http://biostatistics.oxfordjournals.org}.
\url{http://biostatistics.oxfordjournals.org}.


\section*{Acknowledgments}

The authors thank XXX

Funding for the project was provided by BIDS.
{\it Conflict of Interest}: None declared.


\bibliographystyle{DeGruyter}
\bibliography{refs}


\begin{figure}[!p]
\centering
%\includegraphics{fig1}
\caption{}
\label{Fig1}
\end{figure}

\end{document}
