%\documentclass[2columns]{article}

\documentclass[letterpaper,12pt]{article}
\usepackage{url}
\usepackage{graphicx}
\usepackage{url}
\usepackage{wrapfig}
\usepackage[normalem]{ulem}
\usepackage{float}
\usepackage{xr}

\usepackage{amsmath,amssymb,amsfonts,amscd,latexsym}
\usepackage{tabularx}
\usepackage{multirow}
\usepackage{multicol}
\usepackage{eurosym}
\usepackage{geometry}
\usepackage{fancyhdr}
\usepackage{enumitem}
\usepackage{xspace}

\usepackage{booktabs}
\usepackage[svgnames,table]{xcolor}
\usepackage[tableposition=above]{caption}
\usepackage{pifont}

\newcommand*\CHECK{\ding{51}}

% Two column foonotes
\usepackage{dblfnote}


%% This is the recommended preamble for your document.

%% Load De Gruyter specific settings 
\usepackage{dgjournal}          

%% The mathptmx package is recommended for Times compatible math symbols.
%% Use mtpro2 or mathtime instead of mathptmx if you have the commercially
%% available MathTime fonts.
%% Other options are txfonts (free) or belleek (free) or TM-Math (commercial)
\usepackage{mathptmx}

%% Use the graphics package to include figures
\usepackage{graphics}

%% Use natbib with these recommended options
%\usepackage[authoryear,comma,longnamesfirst,sectionbib]{natbib} 
\usepackage{natbib}


%\externaldocument{supp}


\newcommand{\todo}[1]{\textbf{[TODO: #1]}}
\newcommand{\note}[1]{\textbf{[NOTE: #1]}}
\newcommand{\fixme}[1]{\textbf{[FIXME: #1]}}

\newcommand{\OMIT}[1]{}
\newcommand{\Xb}{\textbf{X}}
\newcommand{\RR}{\mathbb{R}}
\newcommand{\Dcal}{\mathcal{D}}
\newcommand {\br}[1]{\left(#1\right)}

\begin{document}

% Title of paper
\title{The art of modelling the 3D structure of the genome\\
  Insights in {\em P. falciparum} genome organization}
\author{Nelle Varoquaux}

%\maketitle


\begin{abstract}

\end{abstract}


\section{Introduction}
\label{sec:introduction}

With more than 217 millions cases nearly 429 000 deaths in 2015,  malaria
remains a major disease burden in tropical and subtropical countries and an
impediment to economic development. In Africa, where more than 90\% of the
cases and deaths occurs, the effects of malaria extend far beyond direct
measures of mortality and the disease is thought to cost more than US\$21
billion a year \citep{onwujekwe:do}.
Malaria is caused by small unicellular protozoan {\em Plasmodium}, infecting
its host through a mosquito bite. Out of the 5 species that can infect human,
{\em P. falciparum} is by the far the most deadly with 99\% of the deaths
associated to it. The parasite has a complex life cycle, with multiple stages
both in the human and mosquito host.


\begin{figure}
\caption{The life cycle of {\em P. falciparum}}{
\small
The human host is infected by \textit{sporozoites} through the bite of a infected female  {\em Anopheles}
mosquito. The \textit{sporozoites} quickly migrate to the liver, where they
start a two weeks multiplication process. They are then released as
\textit{merozoites} in the bloodstream and proceed to infect red blood cells. The
parasites then start their ``erythrocytic'' cycle (through \textit{rings},
\textit{trophozoites} and \textit{schizonts} stages), via another round of
replications in red blood cells, through an unusual process of cell division
called \textit{schizogony}. The parasite first undergoes multiple rounds of nuclear
replication and division into 13 to 32 daughter cells, until the red blood
cell bursts and provokes the release of \textit{merezoites} and the infective cycle
starts anew. This asexual replication cycle is responsible for the symptoms
and the complications of the disease: anemia, tertian fever, \dots}
\end{figure}

While efficacious vaccines still remains a hope and drug resistance to anti
malaria continues to arise, genomic-based research on malaria seeks new
avenues for developing novel therapies \citep{kirchner:recent}. One of the
limiting factors of developments of new drugs is our poor understanding of the
mechanisms underlying the parasite's complex life cycle. While the development
of {\em P. falciparum} through the different stages of its life is thought to
be driven by coordinated changes in gene expression, the relative paucity of
transcription factors points to unusual gene regulatory mechanisms. Meanwhile,
the relative abundance of proteins related to chromatin structures, mRNA decay
and translation rates suggests alternative mechanisms of gene regulation at
the epigenetic and post-translational levels \citep{cui:chromatin-mediated,
duffy:role, hoeijmakers:placing, horrocks:control, deitsch:mechanisms}:
a better understanding of genome architecture of {\em P. falciparum}, both at
a local and global scale may provide new leads for developing new therapies.

In recent year, chromosome conformation captures-like method, broadly referred
to as Hi-C allowed to identify physical interactions between two regions in a
genome-wide fashion, yielding information on their relative spatial distance
in the nucleus. Hi-C has opened new avenues for more systematic analysis of
the three-dimensional folding of the genome, paving the way for a better
understanding of the relations between 3D structure and gene regulation,
replication timing, epigenetic changes as well as many other biological
processes. The 3D genome of a wide variety of organisms has been studied in
the past year, including many species of yeasts \citep{duan:three-dimensional,
burton:species-level, mizuguchi:cohesin-dependent}, bacteria
\citep{umbarger:three-dimensional}, flies \citep{sexton:three-dimensional},
plants \citep{feng:genome-wide, wang:genome-wide} and numerous human and mouse
cell lines \citep{lieberman-aiden:comprehensive, rao:3D}. Within the past few
years, two studies of the three-dimensional structure of {\em P. falciparum}
have been released.  \citet{lemieux:genome-wide} probed several strains of
{\em P. falciparum} to understand the link between 3D structure and the
complex regulation of \textit{var} genes, a family of genes involved the
invasion of red blood cell and also responsible for the parasite's great
capacity to evade our immune system. \citet{ay:three-dimensional} assayed the
3D structure of the genome of the parasite during three key stages of the
erythrocytic cycle. The relatively small size of the genome yet relatively
complicated genome architecture, the complex life cycle and the strong link
between chromatin folding, gene regulation and epigenetic places {\em P.
falciparum} as an interesting case study for developing new methods for better
understanding the properties of genome folding and its link to gene
regulation.

An important part of analyzing the three-dimensional structure of the genome
of {\em P. falciparum} (as well as many other organisms) relied on the
inferring accurate 3D models of the structure. These models were then used as
a preliminary step to various analysis, such as identifying colocalized
elements, distinguishing between open and closed chromatin and last but not
least, integrating with other sources of data, such as gene expression or
chromatin modification. In recent years, a profusion of methods were developed
for creating 3D models of the genome, either as standalone methods or with the
specific goal to better understand an organism, a biological process (such as
the inactivation of chromosome X) or the folding of a chromosome. These
methods broadly fall into two categories: ``model-based'' and ``data-driven''.
The former methods consider the polymer nature of DNA to leverage the
theoretical and computational work done in statistical physics of polymers to
build with as few assumptions as possible many chromosome conformations. Those
chromosome conformations are then used to compare against experimental data,
such as Hi-C contact count matrices, in order to iteratively improve the
models. These models offer mechanistical insights into the folding of DNA. The
latter approaches use the experimental data to infer 3D models, by typically
minimizing a cost function ensuring the models are as consistent as possible
with the data.

This paper provides a view on the work accomplished on building and using 3D
models using contact maps for better understanding of genomic and epigenetic
processes, with a particular highlight on the {\em P falciparum}. It dwells on
modeling challenges: why, how and what do 3D models teach us about gene
regulation? The first and second sections of this paper discuss respectively
current data- and model-driven methods to build 3D structures of DNA
using contact maps.  The last section reviews how these models can be used to
discover the key role of genome architecture in biological process, with a
particular focus on discoveries on {\em P. falciparum} genome architecture.

% XXX See rosa:computational

\begin{figure}
\centering
\includegraphics[width=0.5\linewidth]{schemas/schema.pdf}
\caption{The art of understanding 3D genome structure}{
\fixme{write caption}}
\label{Fig1}
\end{figure}


\section*{Three-dimensional models of DNA from Hi-C data}

The development of genowe-wide and high-throughput protocols to probe samples
for their 3D genome architecture naturally paved the way to systematic and in
depth study of the folding mechanisms of DNA. Over the past 10 years, the
challenge of building 3D models from contact maps have been tackled by a
plethora of methods, some developped for a particular organisms, others with
the hope to be generalizable to any data sets. Methods model chromosomes as a
series of beads, and attempt to place the beads in a 3D euclidean space to
well represent the contact map. ``data-driven'' methods broadly fall into two
categories: (i) \textit{consensus} approaches, that aim at inferring a unique
mean structures best representing the contact count data; (ii)
\textit{ensemble} methods that yield a population of structures.

Both consensus and ensemble methods have benefits and drawbacks. Ensemble
approaches are more biologically accurate: Hi-C data are derived from a
population of cells, each of those with a uniquely folded 3D structure. The
variability of structures among ours cells may thus be better represented by a
population of structures. Yet, in addition of being more complex and
computationally intensive, ensemble approaches raise the question of
interpretability. Often, one has to fall back to interpreting the mean
structure \citep{kalhor:genome}, or a reduced set of structures
\citep{rousseau:three}. On the other hand, consensus approaches yield a single
structure recapitulating the rich information provided by Hi-C data, including
hallmarks of genome architecture shared across all cells despite the
cell-to-cell variability \citet{nagano:single-cell}. More amenable for
visualization and analysis, this average structure can be easily integrated
with other sources of data, such as RNA-seq, chip-seq etc, which are also
population-based.


\begin{figure}
\centering
\includegraphics[width=0.7\linewidth]{figures/counts_maps.png}
\caption{Contact maps of {\em P. falciparum}'s chr7}{
\fixme{write caption}
\textbf{A.} \citep{lemieux:genome-wide}}
\label{Fig1}
\end{figure}

\subsection*{Consensus models}

Consensus methods aim at inferring a model $\mathbf{X} \in \RR^{3 \times n}$
that best represents the contact count matrix $\mathbf{X} \in \RR^{3 \times
n}$, usually through the minimization of an objective function
$\mathcal{O}(\mathbf{O}, \mathbf{C})$, sometimes under constraints.

\begin{equation*}
\renewcommand{\arraystretch}{2}
\begin{array}{ccl}
\underset{\Xb}{\text{minimize}} & & \mathcal{O}(\mathbf{C}, \Xb) \,,
\end{array}
\end{equation*}

The objective function $\mathcal{O}$, sometimes called the ``scoring
function'', can be derived from known embedding algorithms (such as
multidimensional-scaling methods), from statistical modeling of contact
counts, or simply constructed in an \textit{ad hoc} manner to fulfill a set of
wished properties.

\subsubsection*{Metric MDS-based methods}

Early methods cast the 3D inference problem as a
multidimensional scaling (MDS) problem: chromosomes are models as a chain of
beads, to place in 3D such that the distance between bead $i$ and $j$ matches
as closely as possible a ``wish-distance'' $\delta_{ij}$ derived from contact
counts \citep{dekker:capturing, duan:three-dimensional,
tanizawa:mapping, ay:three-dimensional} .

\begin{equation*}
\renewcommand{\arraystretch}{2}
\begin{array}{ccl}
\underset{\Xb}{\text{minimize}} & & \underset{i, j \in \mathcal{D}}{\sum}
\frac{1}{w_{ij}} (d_{ij} - \delta_{ij})^2 \\
\text{subject to} & & \text{Biologically motivated constraints} \\
\end{array}
\end{equation*}


The first step is thus to find an adequate count-to-distance mapping to
convert contact counts into pairwise wish-distances $\delta_{ij}$. 
\citet{dekker:capturing} models the 78 pairwise contact count of {\em S.
cerevisiae}'s chromosome III as a Worm-like chain polymer,
\citet{duan:three-dimensional} uses a linear mapping between contact counts
and wish distances, \citet{ay:three-dimensional} relies on the biophysical
properties of fractal globule polymers, \citet{tanizawa:mapping} inferred the
count-to-distance mapping by fitting the relationship using known pairwise
distances obtained through high-resolution FISH measures. In brief, there are
as many ways to derive the count-to-distance mapping as there are of studies.

\citet{lesne:3d} proposes a different approach to convert counts into
distances: they borrow the concept of shortest path from graph theory to
create a distance matrix. A graph is constructed from the contact map by
considering each loci as a node. Loci seen interacting are connected through
an edge of weight $\frac{1}{c_{ij}}$. The authors then compute wish-distances
$\delta_{ij}$ as the shortest path between node $i$ and $j$ in the graph.
Constructing wish-distances in such a way has two advantages: (i) low contact
counts do not contribute much to the wish-distances; (ii) the resulting
wish-distances form a distance matrix, and thus an optimal solution to the MDS
can be found.

In addition of having subtle differences in the derivation of wish distances
(which can have important effects on the resulting structures),
the different methods also vary by the inclusion of weights to reflect
confidence in ``wish-distances'' \citep{ay:three-dimensional}, or constraints
to reflect prior knowledge on the structure. Table~\ref{table:mds_detail}
summarizes the characteristic of each method.

\begin{table*}
\scriptsize
\centering
\begin{tabular}{rccccccccc}
\hline
\multirow{2}{*}{\textbf{Paper}} & \multirow{2}{*}{\textbf{Organism}} &
\textbf{Full or}
& \textbf{Counts-to-distance} &
\multicolumn{4}{c}{\multirow{2}{*}{\textbf{Constraints}}}
& \multirow{2}{*}{\textbf{Weights}}\\
 & & \textbf{partial} & \textbf{mapping} &  \\
 \cmidrule(lr){5-8} 
 & & & & Adj. & rDNA & C & T & \\
\hline
\cite{dekker:capturing} & {\em S. cerevisiae} & Chr III & Worm-like chain
behavior & & & & \\
\cite{duan:three-dimensional} & {\em S. cerevisiae} & Whole-genome & Linear
relationship & \CHECK &  \CHECK & \CHECK &  & \\
\cite{tanizawa:mapping} & {\em S. pombe} & Whole-genome &
FISH-derived relationship & \CHECK & \CHECK & \CHECK & \CHECK &  \\
\cite{ay:three-dimensional} & {\em P. falciparum} & Whole-genome & Fractal
globule derived relationship & \CHECK  & & &
& $\frac{1}{\delta_{ij}^2}$ \\
\cite{peng:sequencing} & {\em P. falciparum} & Whole-genome & Fractal
globule derived relationship & \CHECK  & & &
 & $\frac{1}{\delta_{ij}^2}$ \\
\cite{lesne:3d} & & Whole-genome & \textit{ad hoc} & & & & & \
\end{tabular}
\caption{Differences between MDS-based methods}{}
\label{table:mds_detail}
\end{table*}


\subsubsection*{Non-metric MDS-based method}

A crucial step of MDS-based method is the conversion of counts into
wish-distances. As described earlier, converting contact counts into
wish-distances require strong assumptions that may not be met in practice. For
example, this mapping changes from one organism to another
\citep{fudenberg:higher-order}, from one resolution to another
\citep{zhang:inference} or even from one time point to another during the cell
cycle \citep{le:high-resolution, ay:three-dimensional}. To alleviate this
problem, after filtering interaction counts based on significance and
interpolation of missing values, \citet{ben-elazar:spatial} casts the
inference as a \emph{non-metric MDS} \citep{kruskal:multidimensional}, where
the 3D structure is inferred jointly with the wish-distances. Another idea is
to parametrize the count-to-distance mapping as a power-law ($d = \beta
c^\alpha)$, and to infer jointly the parameters jointly with the structure
\citet{zhang:inference}.

\subsubsection*{Other ad-hoc optimization methods}

A series of methods cast the inference as an optimization method, where the
objective function is designed by hand to fulfill a number of properties
\citep{trieu:large, trieu:MOGEN, trieu:3D}. For example, \citep{trieu:large}
formulates an optimization problem that attempts to place the beads such that,
if a significant contact is observed between $i$ and $j$, then $\| x_i - x_j
\| < d_c$. On the other hand, if a no significant contact is observed, then
the two beads should be no closer than $d_c$. In addition, the authors also
place constraints on the minimum distance separating two beads and maximum
distances between two beads. To satisfy all those properties, the authors
propose to minimize the following objective function:

\begin{equation*}
\begin{aligned}
\mathcal{O}(X, \delta_{ij}) = & \underset{i, j | c_{ij} \neq 0, |i-j| \neq 1}{\sum} \Bigg(W_1 \text{tanh}(d^2_c -
d^2_{ij}) c_{ij} + W_2 \frac{\text{tanh}(d^2_{ij} - d^2_\text{min})}{N} \Bigg) \\
& \underset{i, j | c_{ij} = 0}{\sum} \Bigg(W_3 \frac{\text{tanh}(d^2_\text{max} -
d^2_{ij})}{N} + W_4 \frac{\text{tanh}(d^2_{ij} -
d^2_{c})}{N}\Bigg) \\
& \underset{i, j | |i - j| = 1}{\sum} \Bigg(W_1 \frac{\text{tanh}(d^2_\text{amax} -
d^2_{ij})}{N} + W_2 \frac{\text{tanh}(d^2_{ij} -
d^2_\text{min})}{N}\Bigg) \,,\\
\end{aligned}
\end{equation*}

where $N$ is the total number of interaction, $(W_1, W_2, W_3, W_4)$ weights
associated to each term, $d_\text{amax}$ the maximum distance separating two
adjacent beads, $d_\text{min}$ the minimum distance separating two adjacent
beads and $d_\text{max}$ the maximum distance between two beads. 
This method is then generalized to include \textit{trans} contact counts data
\citep{trieu:MOGEN}

To give another example of carefully crafted optimization, \cite{trieu:3D}
proposes to exploit the Lozentzian function, to formulate a constraint-based
problem. They first convert the contact counts $c_{ij}$ into wish-distances,
and cast the following optimization problem:

\begin{equation*}
\renewcommand{\arraystretch}{2}
\begin{array}{ccl}
\underset{\Xb}{\text{maximize}} & & \underset{| i - j | = 1}{\sum} \frac{c^2
c_\text{max}}{c^2 + (d_{ij} - \delta_{ij})} + 
\underset{| i - j | \neq 1}{\sum} \frac{c^2 c_{ij}}{c^2 + (d_{ij} - \delta_{ij})}
\end{array}
\end{equation*}

This objective problem attempts to minimize the difference between the
wish-distance $\delta_{ij}$ and the euclidean distance $d_{ij}$ between bead
$i$ and $j$ similarly to MDS-based method, but exploiting the property of the
Loretzian function to have very high derivative when constraints are broken.

\fixme{maybe skip this section}


\subsubsection*{Statistical-based models}

\citet{varoquaux:statistical} proposes a novel approach called \texttt{Pastis}
based on statistical modeling of contact counts as random Poisson variables
where the intensity is a function of the distance: $c_{ij} \sim
\text{Poisson}(\beta d_{ij}^\alpha)$, and some perform the optimization only
including a subset of contact counts. The authors propose to cast the
inference of 3D models as maximizing the likelihood:

\begin{equation}
\renewcommand{\arraystretch}{2}
\begin{array}{cll}
\underset{\alpha, \beta, \textbf{X}}{\text{max}} &
\mathcal{L}(\mathbf{X}, \alpha, \beta) = \underset{i<j\leq n}{\sum}  c_{ij}
\alpha \log d_{ij} + c_{ij} \log \beta - \beta d_{ij}^\alpha &\\
\end{array}
\end{equation}


\texttt{Pastis} can automatically adjust the parameters $\beta$, $\alpha$ of
the counts-to-distance transfer function and infer a genome structure that
best explains the observed data. The strength of this methods comes from the
robustness to low signal-to-noise ratio, through a direct modeling of the
noise through the statistical model.

\fixme{add results on the P. falciparum to show pastis's results on 
super noisy data}

\subsection*{Ensemble methods}

{\em Ensemble approaches} aim at inferring a population of structures
representative of the contact count map. The methods fall into two distinct
categories: the first type of problem casts a restraint-based 
optimization problem and sample local minima of the function
\citep{bau:three-dimensional, umbarger:three-dimensional}, while the second
type proposes a statistical modeling of the problem and samples the posterior
distribution \citep{rousseau:three, hu:bayesian}. A parallel can be made
between the first group of methods and {\em Consensus} MDS type of approaches,
and the second with {\em Consensus} statistical based methods.

\subsubsection*{Sampling local minima}

\citet{umbarger:three-dimensional,bau:three-dimensional, kalhor:genome} models
chromosomes as a series of beads, linked by restraining oscillators. These
oscillators can be thought of as a ``force'' between beads so that they come
into contact or ensure a minimal or maximal distance between those. The models
includes three differente types of restraints: (i) beads seen interacting are
restrained with harmonic oscillators of strengths derived from the contact
counts; (ii) adjacent beads are ensured to be neither too close nor too far
from one another; (iii) the last oscillators are activated only when
constraints are not fulfilled to XXX.  This yields an optimization problem
with a large number of local minima, which the authors sample from by running
50,000 minimizations starting from random starts.

\subsubsection*{Estimating the posterior distribution}

Similarly to \citet{varoquaux:statistical}, \citet{rousseau:three} and
\citet{hu:bayesian} proposes to model contact counts with a formal
probabilistic model. \citet{rousseau:three} models observed contact counts
$c_{ij}$ as a random Gaussian variable of mean $\beta d_{ij}^{\alpha}$,
$\alpha \leq 0$ and variance $\sigma_{ij}$ estimated directly from the contact
count data, while \citet{hu:bayesian} models contact counts as random Poisson
variables of mean $\beta d_{ij}^\alpha$. The authors then sample from the
posterior using MCMC. Obtaining a consensus structure from such a method can
be accomplished by selecting the maximum a posteriori.

\subsection*{Single-cell models}

The last category of data-driven methods to infer the 3D architecture of the
genome rely on a new protocol to probe single-cell for their 3D structures
\citep{nagano:single-cell,ramani:massively}. Single-cell Hi-C is still in its
early days, and, despite potential for assessing the variability of
cell-to-cell genome architecture in a genome-wide fashion, only a handful of
data sets are today publicly available. The contact maps originating from
these data sets are very sparse, and specific methods to infer 3D structures
need to be developed specifically for sc-HiC. Two methods have been developed
to tackle the challenging task of constructing models from this data.

Akin to {\em consensus} MDS-like method and {\em ensemble}-local minima
methods,  the first approach is to consider each contact as a constraint and
to formulate an under constrained optimization problem. A population of
structures satistifying the constraints can be found by sampling local
minimas.

\citet{paulsen:manifold} proposes a very different method, and formulate a
\textit{manifold based optimization}, where a low rank PSD matrix (and thus a
distance matrix) is optimized to be as close as possible to the sparse contact
count matrix. Applying classical MDS on this low rank PSD matrix then yields a
\textit{consensus} 3D model of the genome.

\begin{table}[ht!]
\begin{center}
\scriptsize
\begin{tabular}{rlccccc}
\hline
\multirow{2}{*}{\textbf{\footnotesize Publication}} & \multirow{2}{*}{\textbf{\footnotesize Name}} &
\textbf{\footnotesize Consensus or} 
&\multirow{2}{*}{\textbf{\footnotesize MDS-based}} & \textbf{\footnotesize Statistical} &
\multirow{2}{*}{\textbf{\footnotesize Available}} \\
& & \textbf{\footnotesize Ensemble}
 & & \textbf{\footnotesize model} & \\
\hline
\hline
\scriptsize{\cite{dekker:capturing}} & & C & \CHECK & & \\
\scriptsize{\cite{duan:three-dimensional}} &  & C & \CHECK & & \CHECK \\
\scriptsize{\cite{tanizawa:mapping}} & & C & \CHECK & & \\
\scriptsize{\cite{ay:three-dimensional}} & & C & \CHECK & & \\
\scriptsize{\cite{ben-elazar:spatial}} & & C & \CHECK & & \CHECK \\
\scriptsize{\cite{varoquaux:statistical}} & Pastis & C & & \CHECK & \CHECK\\
\scriptsize{\cite{bau:three-dimensional}} & & E & & &  \\
\scriptsize{\cite{umbarger:three-dimensional}} & & E  & & &\\
\scriptsize{\cite{zhang:inference}} & chromSDE & C &  \CHECK & & \CHECK\\
\scriptsize{\cite{peng:sequencing}} & autochrom3D & C &  \CHECK & & \CHECK\\
\scriptsize{\cite{rousseau:three}} & & E & & \CHECK & \CHECK\\
\scriptsize{\cite{hu:bayesian}} & Bach & E/C &  & \CHECK & \CHECK\\
\scriptsize{\cite{kalhor:genome}} & & E &   & &\\
\scriptsize{\cite{lesne:3d}} & ShRec3D & C  & \CHECK & & \CHECK \\
\scriptsize{\cite{trieu:large}} & & C & & & \\
\scriptsize{\cite{trieu:3D}} & & C & &  & \\
\scriptsize{\cite{trieu:MOGEN}} & MOGEN & C & & & \\
\scriptsize{\cite{nagano:single-cell}} & & E & & & \\
\scriptsize{\cite{paulsen:manifold}} & & C & \CHECK & & \CHECK \\
\hline
\end{tabular}
\end{center}
\caption{\bf A comparison of 3D inference methods}{\small In this table, we summarize
properties of published methods to infer the 3D structure of the genome: (1)
is it a consensus or a ensemble based inference? (2) Is it an MDS based
method? (3) or relies on a statistical modeling; (4) is the software
available or not (to the best of our knowledge).}
\end{table}

\section*{Model-driven approach: modeling DNA as a flexible fibers under constraints}

``data-driven'' methods, as presented in the last section, use the
experimental contact maps to infer models as consistent as possible with the
data. Another approach consists in modeling chromosomes as polymers under a
small number constraints such that a generated contact maps from these models
match as closely as possible the observed contact maps. These ``model-driven''
approaches offer mechanistical insights into the genome architecture.


\subsection*{A small set of constraints explain the 3D architecture of \textit{S. cerevisiae}}

The budding yeast \textit{S. cerivisiae}'s 3D structure has been extensively
studied, both through 3C-type studies \citep{dekker:capturing,
duan:three-dimensional, burton:species-level} and through bio-imaging experiments
\citep{berger:high}. The small size of its genome, the well-known hallmarks of its
genome architecture and the availability of high resolution contact maps and
FISH data set quickly led several teams to investigate the minimal set of
constraints needed to reproduce the hallmarks of its genome architecture.

\cite{tjong:physical, tokuda:dynamical, tjong:physical} model {\em S.
cerevisiae}'s chromosomes as flexible random fibers under a small set of
constraints. While the exact modeling proposed by the three groups differ, the
set of constraints can roughly be summarized as: \textit{(i)} the chromosomes
are constrained into a spherical ball representing the nucleus; \textit{(ii)}
centromeres are constrained into a spherical ball tethered to the nuclear
membrane; \textit{(iii)} telomeres are tethered to the nuclear membrane;
\textit{(iv)} rDNA is constrained into the nucleolus, represented as a
spherical ball opposite to the centromeres. A scoring function eng


Using IMP, the authors generate a 25,000 structures, all of which respect the
constraints listed above. The population of structures is then used to created
a ``volume-exclusion contact map'', considering that two beads that are less
then 45~nm apart from a contact. The Pearson correlation of the
volume-exclusion contact map and the Hi-C one are highly correlated,
demonstrating this small set of constraints fully explain the observed counts.
In addition, the population of structures also explain FISH experiments
previously published.


\section*{Downstream analysis using 3D models: a highlight of the study of
{\em P. falciparum}'s 3D structure}

In previous section, I have reviewed data-driven methods to infer either
\textit{consensus} or \textit{ensemble} models of the 3D structure. The reader
may well ask why go through the effort to obtain such model and not directly
study the contact maps?  In this section, I will review a number of downstream
analysis one can do on 3D models, highlighting but limiting myself to results
on the 3D structure of {\em P. falciparum}.

\subsection{Probing the 3D structure of the {\em P. falciparum}}
In the past year, two groups have published studies of {\em P.
falciparum}'s 3D genome architecture using Hi-C. \citet{lemieux:genome-wide}
studied several strains associated with populations expression unique
\textit{var} genes, in order to study key folding properties of the 3D
structure relating to \textit{var} gene expression. On the other hand,
\citet{ay:three-dimensional} focus three key time-point of the development
cycle. Table~\ref{tab:data} summarizes the available data sets and their key
properties after processing the data as described in
\citet{ay:three-dimensional} and combining all available libraries for each
life-cycle stage and strain.



\begin{table}
\scriptsize
\centering
\begin{tabular}{cccccccc}
\hline
\multirow{2}{*}{\textbf{Name}} & \multirow{2}{*}{\textbf{Strain}} &
\multirow{2}{*}{\textbf{Stage}} & \multirow{2}{*}{\textbf{Resolution}} &
\textbf{Number} & \textbf{Perc} & \textbf{Perc}& \multirow{2}{*}{\textbf{Reference}} \\
& & & & \textbf{of contacts} & \textbf{of \textit{cis}}& \textbf{of trans}& \\
\hline
\hline
Ay-\textit{rings} & 3D7 & Late Rings & 10~kb & 16711552 & 43\% & 57\% &  \cite{ay:three-dimensional} \\
Ay-\textit{trophozoites} & 3D7 & Trophozoites &10~kb & 56348498 & 53\% & 47\% & \cite{ay:three-dimensional} \\
Ay-\textit{schizonts} & 3D7 & Schizonts & 10~kb & 11652832 & 55\% & 45 \% & \cite{ay:three-dimensional} \\
Lemieux-\textit{A4+} & IT/BC6+ & Rings & 25~kb & 18488252 & 19\% & 81\% & \cite{lemieux:genome-wide} \\
Lemieux-\textit{A4} & IT/3G8 & Rings &  25~kb & 19674672 & 28\% & 72\% & \cite{lemieux:genome-wide}\\
Lemieux-\textit{A44} & IT/BC6- & Rings & 25~kb & 18660594 & 25\% & 75\% & \cite{lemieux:genome-wide}\\
Lemieux-\textit{DCJ\_On} & NF54/DCJ on & Rings & 25~kb & 3098370 & 26\% & 74\% &\cite{lemieux:genome-wide} \\
Lemieux-\textit{DCJ\_Off} & NF54/DCF Off & Rings & 25~kb & 2533470 & 26\% & 73\% &\cite{lemieux:genome-wide} \\
Lemieux-\textit{B15C2} & NF54/B15C2 & Rings & 25~kb &  1022996 & 12\% & 88\% & \cite{lemieux:genome-wide}\\
\hline
\end{tabular}
\caption{Summary of the available {\em P. falciparum} Hi-C datasets}
\end{table}



\fixme{Add the fact we have time points}

\subsection*{Model evaluation}

A substantial difficulty in modelling the 3D structure of the genome is that
model evaluation tends to the subjective. What is the relevant measure?
``Truth'' is generally not fully available, except for a few pairs of loci or
in simulations. Is validating the colocalization of a pair or a few pairs of
loci via FISH experiments enough? Is fit to the contact maps or agreements
between modeling techniques relevant? 

\fixme{Add section on model evaluation}



\subsection*{Structure stability across time points, clustering and other
variance analysis \dots}

The reader may well ask how sensitive the result is either initialization or
how. In short, it is not. To study the stability of the results with respect
to the initialization, I perform for each stage presented here 1000
optimization, starting from different initialization. 

The first question that arise is simply: are structures from the same time
points but from a different start point more alike than structures from
different time points? To answer this question, I compute for each structure
and each time point a feature matrix, composed of a subsampled pairwise
euclidean distances of each structure (so that the resulting feature matrix
fits in memory on a laptop). I then compute a stochastic PCA on this feature
matrix and plot the projection in 2D space using the 3 first components. The
results clearly show that structures from the same time point cluster
strongly. Repeating this experiment on singly chromosomes yields the same
result.

The second question one can ask is how do the structures differ. Tackling this
question is very challenging, but can be reformulated in a much easier way.
Are the hallmarks of interest conserved across structures of a same stage?
\citet{ay:three-dimensional} and \citet{lemieux:genome-wide} both identified
{\em P. falciparum} folded in very specific ways, with VRSSM genes highly
interacting. \citet{ay:three-dimensional} also observed strong clustering of
the centromeres, and enrichment in interaction at the telomeres. These
observation can lead to a rigorous approach at identifying whether families
of loci clustered in the structures and even quantifying this in a rigourous
manner.

\subsection*{Chromatin compaction}

Bau et al


\subsection*{3D gene set enrichment}

To assess whether groups of genes are colocolized in a 3D model,
\citet{ay:three-dimensional} leveraged a statistical method developed by
\citet{witten:assessment}, which requires labeling each pair of loci in two
groups: "close" or "far".The authors used varying distance thresholds (10\%,
20\% and 40\% of the nuclear diameter) to deem a locus pair “close” and
labeled all remaining pairs in the set as “far”.  The authors then compare the
enrichment of loci pairs of a group being close and far by resampling loci
among a same chromosome.

This approach dichotomizes loci pairs into two groups, and check for the
enrichment of a label in one of the two groups. \citep{capurso:distance-based}
presents an approach that avoids this step, and instead directly estimates the
significance within the 3D model. Briefly, for a group $\mathcal{G}$, MPED
computes a test statistics:
\begin{equation*}
M = \underset{i,j \in \mathcal{G}| c_i != c_j}{\text{median}} d_{ij}\,,
\end{equation*}

where $d_{ij}$ is the euclidean distance between bead $i$ and bead $j$. The
null distribution is estimated empirically by resampling $10^5$ with
preservation of the chromosome structure. If the $M$ statistics is smaller
than the mean of the null distribution, it is compared to the lower tail of
the distribution and indicates co-localization. If the $M$ statistics is
larger than the mean of the null distribution, then it is compared to the
upper tail of the distribution, and indicates dispersion.

Applying MEPD to the Ring stage, \citet{capurso:distance-based} confirm that
centromeres, telomeres, VRSM genes (both overall, subtelomeric and internal)
colocalize.


\subsection*{Integrative analysis of gene expression and 3D structure using KernelCCA}

Last but not least, an exciting contribution of \citet{ay:three-dimensional}
is the integrative analysis of the 3D structure with gene expression using an
unsupervised learning technique called ``kernel Canonical Correlation
Analysis'' \citep{bach:kernel}. In this particular case,  KernelCCA aims at
extracting a set of orthogonal gene expression component that exhibit
coherence with respect (here) to the 3D structure. It can be helpful to
think of this procedure as performing a Principal Component Analysis on the
set of gene expression profiles but such that the component extracted are
correlated with the gene expression.

Let us take a closer look at how formally kCCA is used in this context.
Consider the set of $n$ genes $g \in \mathcal{G}$. Each gene $g$ is
represented on the one hand by its gene expression profile $e(g) \in \RR^{p}$
and its 3D position $x(g) \in \RR^3$. Assume the set of gene expression
profiles is mean centered and of unit variance.

The goal of the procedure is to extract a gene expression component $v\in
\RR^p, \|v\|=1$ such that it is both representative of set of gene expression profile,
but also correlated with the 3D structure. We thus construct two scores to
assess those properties. First, define $V(v) = \underset{g \in
\mathcal{G}}{\sum} (v^T e(g))^2$. $V(v)$ measures the percentage of variance
explained among the gene expression profile when projected onto the component
$v$. Note that maximizing this score results in finding the first principal
component. To assess smoothness along the structure,
\citet{ay:three-dimensional} leverage a standard approach in kernel methods,
where the smoothness of a score $f$ is quantified by the function:

\begin{equation}
S(f) = \frac{f^TK^{-1}_{3D}f}{\|f\|}\,,
\end{equation}

where $K_3D$ is the Gaussian kernel matrix of the genes' 3D coordinates. The
smaller $S(f)$, the smoother $f$ is with respect to the 3D structure.

In summary, to obtain smooth gene expression components $v$ on the 3D
structure representative of the variance amongst all gene expression, one
needs to: (1) maximize $V(v)$; (2) minimize $S(f)$; (3) maximize the
correlation between $f$ and $v^T e(g)$. In other word, one would need to solve
the kernelCCA problem, using on one hand a linear kernel over the gene
expression profile, while on the other the Gaussian kernel over the 3D
coordinates of each genes.

\citet{ay:three-dimensional} apply this method on the three model obtained,
and find highly correlated gene expression profiles with the 3D structure.
Ranking the genes with their projection onto the gene components,
\citet{ay:three-dimensional} demonstrate that several gene families and GO
terms are enriched both close to the telomeres and at the opposite end of the
nucleus.



\section*{Discussion}

\section*{Acknowledgments}



{\it Conflict of Interest}: None declared.


\bibliographystyle{DeGruyter}
\bibliography{refs}


\end{document}
